### PR TITLE
Add Support for Rails 7.0 & Rails 7.1

### PR DESCRIPTION
**Pull request by mkhairi**

### CODE REVIEW

**Comment by mkhairi** [comment]
Hi @marshall-lee, Hope you're well. Could you kindly review this PR when you have a moment? Thanks.

**Comment by marshall-lee** [comment]
Hi @mkhairi please fix the CI.

**Comment by marshall-lee** [comment]
@mkhairi Lets drop Ruby 2.5 support, it's way too old.

File: lib/nested_record/lookup_const.rb [hunk 1]
```diff
@@ -2,17 +2,34 @@
...
+  end
+
+  def self.safe_constantize(type_name)
+    # TODO: Remove this when we drop support for Rails < 7.0
+    if ActiveSupport::Dependencies.respond_to?(:safe_constantize)
+      ActiveSupport::Dependencies.safe_constantize(type_name)
+    else
+      type_name.safe_constantize
+    end
+  end
```
Comment (marshall-lee) [inline lib/nested_record/lookup_const.rb]:
```suggestion
  # TODO: Remove this when we drop support for Rails < 7.0
  if ActiveSupport::VERSION::MAJOR < 7
    def self.constantize(type_name)
      ActiveSupport::Dependencies.constantize(type_name)
    end
    
    def self.safe_constantize(type_name)
      ActiveSupport::Dependencies.safe_constantize(type_name)
    end
  else
    def self.constantize(type_name)
      type_name.constantize
    end
    
    def self.safe_constantize(type_name)
      type_name.safe_constantize
    end
  end
```

**Review by marshall-lee — CHANGES_REQUESTED** [review]

File: lib/nested_record/lookup_const.rb (hunk 1 above)
Comment (mkhairi) [reply lib/nested_record/lookup_const.rb]: @marshall-lee, Opps.. Seems like it works without any rails version condition here. 😅

Comment (marshall-lee) [reply lib/nested_record/lookup_const.rb]: @mkhairi Yes it works but `ActiveSupport::Dependencies.constantize` on older Rails leverages the `ActiveSupport::Dependencies::ClassCache`. We should do it too.

Comment (mkhairi) [reply lib/nested_record/lookup_const.rb]: > @mkhairi Yes it works but `ActiveSupport::Dependencies.constantize` on older Rails leverages the `ActiveSupport::Dependencies::ClassCache`. We should do it too.

@marshall-lee, Apologies,  I'm unaware of about this. I have reinstated the Rails version condition based on the previous suggestion.

Comment (marshall-lee) [reply lib/nested_record/lookup_const.rb]: @mkhairi Sure, please squash the commits

**Review by marshall-lee — APPROVED** [review]

**Pull request merged**